### ISSUE TITLE
Wishlist: identity table and its use

### ISSUE THREAD

**Issue by clbr** [cc65/cc65#583]
An identity table is a technique invented by tokumaru that allows emulating instructions like txy, tyx, adx, and ady. 256 bytes, from 0 to 255.

txy: ldy identity,x
tyx: ldx identity,y
adx 5: txy; ldx 5+identity,y

If the compiler supported such, perhaps with a new -O option given it does add 256 bytes, it would enable better code in many situations.

**Comment by spiro-trikaliotis** [comment]
But a table with 256 entries would not be enough for your last case (adx 5). You would need almost the double of space, wouldn't you? (Think about adx $ff with X being $FF beforehand).

Or do I miss something here?

**Comment by clbr** [comment]
Yes, it would only be possible when the compiler can prove it won't wrap. For the uses I had in mind, like iterating over an array, you don't have to worry about wrapping when the array is small enough.

**Comment by MonteCarlos** [comment]
It's a two liner using .rep and .byte. Maybe it would make sense to add this and some other table macros using .macpack?

**Comment by clbr** [comment]
.macpack would only allow user use, no? The goal is for the compiler to generate better code.

**Comment by MonteCarlos** [comment]
in many occasions txa tay would do and if not, a stx tmp ldy tmp seq would have a two cycle penality but no 256 byte table.
